### PR TITLE
feat: allow multiple open modals

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

... by sharing modals state over a proper single {[k: ModalKey]: ModalContext} state, vs. a single ModalContext state and many "independant" contexts previously, which were not working as they should as only one context could live at a given time, and one would override the other.

Effectively fixes the send modal Ledger Open App ack.

## Issue (if applicable)

N/A

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low - do a quick sanity check of all modals across the app in addition to the actual Ledger ack fix.

## Testing

- Do a smol regression test of modals across the app and confirm they still open/close fine
- Confirm Ledger Open App ack is now happy in the send modal

### Engineering

- ^

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

- ^

## Screenshots (if applicable)

image
image

### CODE REVIEW

File: src/context/ModalProvider/ModalContainer.tsx [hunk 1]
```diff
@@ -1,56 +1,225 @@
...
+  manageAccounts: ManageAccountsModal,
+} as const
+
+export const modalReducer = (state: ModalState, action: ModalActions<keyof Modals>): ModalState => {
   switch (action.type) {
     case OPEN_MODAL:
-      return { isOpen: true, props: action.props }
+      return {
+        ...state,
+        [action.key]: { isOpen: true, props: action.props },
```
Comment (gomesalexandre) [inline src/context/ModalProvider/ModalContainer.tsx:213]: The guts of this PR - modals are now indexed per modal in a *global* reducer vs. many reducers (which were useless, as only one reducer was effective)

File: src/context/ModalProvider/types.ts [hunk 2]
```diff
@@ -41,18 +41,21 @@ export type ModalActions<T extends keyof Modals> = OpenModalType<T> | CloseModal
...
+  key: keyof Modals
 }
-
 export type ModalProps<T extends keyof Modals> = React.ComponentProps<Modals[T]>
 
-export type ModalState<T extends keyof Modals> = {
-  props?: ModalProps<T>
-  isOpen: boolean
+export type ModalState = {
+  [K in keyof Modals]: {
```
Comment (gomesalexandre) [inline src/context/ModalProvider/types.ts:55]: The guts of this PR - modals are now indexed per modal in a *global* reducer vs. many reducers (which were useless, as only one reducer was effective)

**Review by NeOMakinG — COMMENTED** [review]
Conceptually looking good, testing now

**Review by NeOMakinG — APPROVED** [review]
Modals are looking like working overall and the send modal is working using the ledger ACK

https://jam.dev/c/e1a83297-85cb-45ee-985b-ddbfba623192

LGTM ✅

File: src/context/ModalProvider/ModalContext.tsx [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import { createContext } from 'react'
 
 import { MODAL_KEYS } from './constants'
-import type { BaseProps, ModalContext, Modals } from './types'
+import type { BaseProps, ModalContextType, Modals, ModalState } from './types'
 
-export const modalContext: ModalContext = MODAL_KEYS.reduce<ModalContext>(
+// @ts-ignore blatant ignore as we inject the correct props during provider creation in ModalContainer
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/context/ModalProvider/ModalContext.tsx]: I like the idea ser! While at it, actually noticed that this copy and paste from the previous implementation wasn't needed anymore: 010e1b73fc7652ff275cd3de413d413d9dbe7876

**Pull request merged**